### PR TITLE
Update podspec and fix MixpanelPrivate

### PR DESCRIPTION
**Pull request by samgreen**

### CODE REVIEW

File: Mixpanel.podspec [hunk 1]
```diff
@@ -14,8 +14,9 @@ Pod::Spec.new do |s|
   s.osx.deployment_target = ''
 
   s.subspec 'Mixpanel' do |ss|
-    ss.source_files  = 'Mixpanel/**/*.{m,h}', 'Mixpanel/**/*.swift'
+    ss.source_files  = 'Mixpanel/**/*.{m,h}', 'Mixpanel/**/*.swift', 'Mixpanel/MPNetwork.{h,m}'
```
Comment (yarneo) [inline Mixpanel.podspec]: what swift files do we have?

Comment (yarneo) [inline Mixpanel.podspec]: do we need MixpanelPeople here too?

File: Mixpanel.podspec [hunk 2]
```diff
@@ -25,15 +26,18 @@ Pod::Spec.new do |s|
   s.subspec 'MixpanelHostWatchOS' do |ss|
     ss.source_files  = 'Mixpanel/**/*.{m,h}', 'Mixpanel/**/*.swift'
     ss.exclude_files = 'Mixpanel/MixpanelWatchOS.{m,h}'
+    ss.private_header_files = 'Mixpanel/MixpanelPrivate.h', 'Mixpanel/MixpanelPeoplePrivate.h'
```
Comment (yarneo) [inline Mixpanel.podspec]: MPNetworkPrivate?

Comment (samgreen) [inline Mixpanel.podspec]: Not used.

File: Mixpanel.podspec (hunk 1 above)
Comment (samgreen) [inline Mixpanel.podspec]: The recursive check should catch everything. MPNetwork could be removed, but linter was giving me trouble.

File: Mixpanel/MixpanelPrivate.h [hunk 3]
```diff
@@ -24,6 +20,9 @@
 #import <CoreTelephony/CTCarrier.h>
```
Comment (yarneo) [inline Mixpanel/MixpanelPrivate.h]: why do we need TARGET_OS_IPHONE inside !MIXPANEL_LIMITED_SUPPORT. If we arent in TVOS and Extensions, only Watch and iOS is left, but Watch doesnt get here right?

**Comment by yarneo** [comment]
LGTM

**Pull request merged**